src/networking.c

```diff
@@ -97,8 +102,7 @@ void linkClient(client *c) {
      * this way removing the client in unlinkClient() will not require
      * a linear scan, but just a constant time operation. */
     c->client_list_node = listLast(server.clients);
-    uint64_t id = htonu64(c->id);
-    raxInsert(server.clients_index,(unsigned char*)&id,sizeof(id),c,NULL);
```

Comment:
i.e. these two mechanisms, each needs primitive functions for adding, removing, and lookup (so that the caller is simple), but i don't think they have to share code.